nameserver/mocks_test.go

```diff
@@ -75,6 +75,10 @@ func (mz *mockedZoneWithRecords) AddRecord(ident string, name string, ip net.IP)
 	notImplWarn()
 	return nil
 }
+func (mz *mockedZoneWithRecords) DeleteRecords(ident *string, ip *net.IP, name *string) int {
```

Comment:
Also I feel the complete opposite about the nil values - it makes it clear that I am not specifying any value at all, rather than a value which is treated specially